python/common_tool_methods.py

```diff
@@ -0,0 +1,146 @@
+import re
+import subprocess
+from datetime import datetime
+from typing import Dict, List
+import os
+
+from github import Repository, PullRequest
+from jinja2 import Environment, FileSystemLoader
+
+from .common_validations import *
+
+
+def get_spec_file_name(project_name: str) -> str:
+    return f"{project_name}.spec"
+
+
+def get_version_number(version: str, fancy: bool, fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}{fancy_suffix}"
+
+
+def get_version_number_with_project_name(project_name: str, version: str, fancy: bool, fancy_release_count: int) -> str:
+    fancy_suffix = f"-{fancy_release_count}" if fancy else ""
+    return f"{version}.{project_name}{fancy_suffix}"
+
+
+def get_template_environment(template_dir: str) -> Environment:
+    file_loader = FileSystemLoader(template_dir)
+    env = Environment(loader=file_loader)
+    return env
+
+
+def find_nth_overlapping(subject_string, search_string, n) -> int:
+    start = subject_string.find(search_string)
+
+    while start >= 0 and n > 1:
+        start = subject_string.find(search_string, start + 1)
+        n -= 1
+    return start
+
+
+def find_nth_overlapping_line_by_regex(subject_string, regex_pattern, n) -> int:
+    lines = subject_string.splitlines()
+    counter = 0
+    for i in range(len(lines)):
+        if re.match(regex_pattern, lines[i]):
+            counter = counter + 1
+        if counter == n:
+            return i
+    return -1
+
+
+def remove_string_inside_parentheses(param: str) -> str:
+    return re.sub(r"[(\[].*?[)\]]", "", param)
+
+
+def run(command, *args, **kwargs):
+    result = subprocess.run(command, *args, check=True, shell=True, **kwargs)
+    return result
+
+
+def cherry_pick_prs(prs: List[PullRequest.PullRequest]):
```

Comment:
I think it gets all the commits I didn't put an extra control

Reply:
>If merge commits should be filtered I can filter it 

Yes, I think we should skip merge commits.

Reply:
When I check the old perl scripts, I do not see any logic to filter out merge commits. It may not be necessary.

See https://github.com/citusdata/tools/blob/develop/automated_packaging/prepare_release.pl

Reply:
I fixed it. I eliminated commits with multiple parents

Reply:
>When I check the old perl scripts, I do not see any logic to filter out merge commits. It may not be necessary.

But including cherry-picking merge commits might bring unwanted ones too ? @hanefi